Retire fix1578 amendment

pratikmankawde opened the pull request:
## High Level Overview of Change
This change retires the `fix1578` amendment.

### Context of Change
Amendments activated for more than 2 years can be retired.
https://xrpl.org/resources/known-amendments#fix1578

### Type of Change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Refactor (non-breaking change that only restructures code)
- [ ] Performance (increase or change in throughput and/or latency)
- [ ] Tests (you added tests for code that already exists, or your new feature included in this PR)
- [ ] Documentation update
- [ ] Chore (no impact to binary, e.g. `.gitignore`, formatting, dropping support for older tooling)
- [ ] Release

pratikmankawde commented on src/test/app/AMMExtended_test.cpp:
```diff
@@ -198,109 +198,100 @@ struct AMMExtended_test : public jtx::AMMTest
+        testAMM(
+            [&](AMM& ammAlice, Env& env) {
+                // Order that can't be filled
+                TER const killedCode{TER{tecKILLED}};
```
Shall I get rid of this(and at other places) `killedCode` variable and use `TER{tecKILLED}` directly in its place?

a1q123456 replied:
I think it's alright

a1q123456 commented:
There's some conflicts. Can you fix them before adding the label please?

bthomee committed:
refactor: Retire fix1578 amendment

Amendments activated for more than 2 years can be retired. This change retires the fix1578 amendment.

bthomee committed:
Merge branch 'develop' into pratik/retire_fix1578_amendment
